lib/ecto/association.ex

```diff
@@ -343,15 +344,21 @@ defmodule Ecto.Association.Has do
     end
 
     on_delete  = Keyword.get(opts, :on_delete, :nothing)
-    on_replace = Keyword.get(opts, :on_replace, :raise)
-
     unless on_delete in @on_delete_opts do
       raise ArgumentError, "invalid :on_delete option for #{inspect name}. " <>
         "The only valid options are: " <>
         Enum.map_join(@on_delete_opts, ", ", &"`#{inspect &1}`")
     end
 
-    unless on_replace in @on_replace_opts do
+    on_replace = Keyword.get(opts, :on_replace, :raise)
+    cardinality = Keyword.get(opts, :cardinality, :raise)
```

Comment:
We should also change this line: https://github.com/eniolopes/ecto/blob/d844662db4c6cd33769ba4a1b5d4174fd02bed5b/lib/ecto/association.ex#L369

It now can use the `:cardinality` variable.